src/scripts/danish-legal/create_completeness_detection.py

```diff
@@ -0,0 +1,299 @@
+from typing import Dict, List, Set
+
+import click
+import pandas as pd
+from datasets import Dataset, DatasetDict, Split
+from docling.document_converter import DocumentConverter
+from huggingface_hub import HfApi
+from requests import HTTPError
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Constants
+COMPLETE_CONTRACT_PROBABILITY = 0.3
+NON_REQUIRED_REMOVAL_PROB = 0.3  # For complete contracts
+OPTIONAL_REMOVAL_PROB = 0.4  # For incomplete contracts
+SYSTEM_PROMPT = (
+    "Identificer om følgende kontrakt er komplet eller om der mangler nogle elementer. "
+    "Beskriv hvilke elementer der mangler."
+)
+
+# Hardcoded mapping to the indices in `contract_sections`
+# for each category.
+REQUIRED_CATEGORIES_TO_INDICES = {
+    "Partner/Identitet": [0],
+    "Arbejdstid": [6],
+    "Ansvarsområde": [2],
+    "Ansættelse": [1],
+    "Ferie/Fravær": [7, 8, 10, 11, 12, 13],
+    "Ophør/Opsigelse": [17],
+    "Løn/Vederlag": [3, 4, 5],
+    "Klausuler": [18],
+    "Personale vilkår": [9],
+}
+
+
+@click.command()
+@click.argument("contract_path", type=click.Path(exists=True, path_type=Path))
+@click.option(
+    "--num-samples",
+    "-n",
+    default=10,
+    help=f"Number of samples to generate (default: {10})",
+)
+def main(contract_path: Path, num_samples: int) -> None:
+    """Create the legal completeness detection dataset and upload it to the HF Hub.
+
+    This script modifies the input contract to create a dataset of samples
+    with complete and incomplete contracts.
+
+    Args:
+        contract_path: Path to the contract PDF file
+        num_samples: Number of samples to generate
+    """
+    # Load and process contract
+    logger.info(f"Processing contract from {contract_path}")
+    if not contract_path.exists():
+        click.echo(f"Error: Contract file {contract_path} does not exist", err=True)
+        return
+
+    converter = DocumentConverter()
+    result = converter.convert(str(contract_path))
+    markdown = result.document.export_to_markdown()
+
+    # Parse contract into sections
+    contract_sections = parse_contract_into_sections(markdown)
+
+    # Get non-required indices
+    not_required_indices = [
+        i
+        for i in range(len(contract_sections))
+        if not any(i in indices for indices in REQUIRED_CATEGORIES_TO_INDICES.values())
+    ]
+
+    # Generate samples
+    logger.info(f"Generating {num_samples} unique samples...")
+    samples = generate_unique_samples(
+        contract_sections=contract_sections,
+        not_required_indices=not_required_indices,
+        num_samples=num_samples,
+    )
+
+    df = pd.DataFrame(samples)
+    dataset = DatasetDict(test=Dataset.from_pandas(df, split=Split.TEST))
+
+    # Remove existing dataset if it exists
+    dataset_id = "EuroEval/legal-completeness-detection"
+    try:
+        api = HfApi()
+        api.delete_repo(dataset_id, repo_type="dataset")
+    except HTTPError:
+        pass
```

Comment:
Hugging Face have now added a new `exists_ok` argument, so that we can avoid these try-except blocks:

```suggestion
    HfApi().delete_repo(dataset_id, repo_type="dataset", exists_ok=True)
```